src/utils/thread_helper.cpp

```diff
@@ -913,8 +914,14 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
     {
         int8_t radioTxPower;
 
-        SuccessOrExit(error = otPlatRadioGetTransmitPower(mInstance, &radioTxPower));
-        wpanStats->set_radio_tx_power(radioTxPower);
+        if ((latestError = otPlatRadioGetTransmitPower(mInstance, &radioTxPower)) == OT_ERROR_NONE)
+        {
+            wpanStats->set_radio_tx_power(radioTxPower);
+        }
+        else
+        {
+            error = latestError;
+        }
```

Comment:
What about we only have one `error` variable and we set it to 'failed' error when any error occurs? This may be easier to understand. The 'lastError` variable name could be ambiguous sometimes.

Also, there could be multiple errors happening in this method and it's not so meaningful to return one precise error when there are many errors. Thus I suggest to use 'fail' as the returned error type.

Reply:
SG. I return the error code OT_ERRROR_FAILED now if there is one or more error(s) happened in the process.